src/main/render/engine.ts

```diff
@@ -17,73 +20,95 @@ function buildArgs({
+  const filter // eslint-disable-next-line function-paren-newline
+  = videoInputs.map((input: VideoInput) => [ '-ss', (startTime + input.startTime).toString(), '-i', input.file ]).flat(1).concat(
+    // eslint-disable-next-line function-paren-newline
+    audioInputs.map((input: AudioInput) => [ '-ss', (startTime + input.startTime).toString(), '-i', input.file ]).flat(1).concat([
+      '-filter_complex',
+      [
+        videoInputs.map((input: VideoInput, i: number) => `[${i}:v]setpts=PTS-STARTPTS,scale=${input.resolution.width}x${input.resolution.height}[input${i}];`).join(''),
+        `${videoInputs.map((__: VideoInput, i: number) => `[input${i}]`).join('')}xstack=inputs=${videoInputs.length}:layout=${videoInputs.map((input: VideoInput) => `${input.position.left}_${input.position.top}`).join('|')}[matrix];`,
+        outputType === 'thumbnail' ? `[matrix]fps=${thumbnailEvery}[pr];` : '',
+        `${outputType === 'thumbnail' ? '[pr]' : '[matrix]'}scale=${outputResolution.width}:${outputResolution.height},setsar=1:1[out]${audioInputs.length > 0 ? ';' : ''}`
+        //audioInputs.map((input: AudioInput, i: number) => `[${i + videoInputs.length}:a]setpts=PTS-STARTPTS,volume=${input.volume || 256}[ainput${i}];`).join(''),
+        //audioInputs.map((__: AudioInput, i:number) => `[ainput${i}]`).join(''), `amerge=inputs=${audioInputs.length}[aout]`
+      ].join(''),
+      '-map', '[out]'
+      //'-map', '[aout]',
+    ]));
+
+  let args: string[];
+
+  if (outputType === 'thumbnail') {
+    args = filter.concat([ 'thumbs/%04d.png' ]);
+  } else {
+    args = [ outputType === 'preview' ? '-re' : 'REMOVED' ].concat(filter).concat([
+      '-c:v', 'libx264',
+      '-c:a', 'aac',
+      '-ac', '2',
+      '-ar', audioSampleRate.toString(),
+      '-x264opts', `keyint=${framesPerSecond}:min-keyint=${framesPerSecond}:no-scenecut`,
+      { 'preview': '-f', 'thumbnail': 'REMOVED', 'render': '-f' }[outputType],
+      { 'preview': 'dash', 'thumbnail': 'REMOVED', 'render': 'mp4' }[outputType],
+      //'-min_seg_duration', '2000000',
+      '-b:v', videoBitRate,
+      '-b:a', audioBitRate,
+      '-preset', outputType === 'render' ? 'medium' : 'ultrafast',
+      '-tune', 'zerolatency',
+      '-maxrate', videoBitRate,
+      '-bufsize', bufferSize,
+      '-v', 'info',
+      '-aspect', aspectRatio,
+      '-vol', outputVolume.toString(),
+      '-metadata', 'description="Made with VideoBones"',
+      '-stats'
+    ])
+      .concat([ outputType === 'render' ? outputFile : previewManifest ]).filter(str => {
+        return str !== 'REMOVED';
+      });
+  }
+
+  console.log(args);
+  return args;
 }
 
 let ffmpeg: ChildProcessByStdio<null, Readable, null> | null;
+let ffmpeg_thumbs: ChildProcessByStdio<null, Readable, null> | null;
 
 export function start(
   options: EngineOptions,
   statusCallback: (elapsedTime: string, donePercentage: number) => void,
   doneCallback: () => void
 ) {
+  if (options.outputType === 'thumbnail') {
```

Comment:
This could be done better, if there is a check anyways, why have separate functions?